cocotb/share/lib/simulator/simulatormodule.c

```diff
@@ -776,14 +815,15 @@ static PyObject *get_type(PyObject *self, PyObject *args)
         return NULL;
     }
 
-    result = gpi_get_object_type((gpi_sim_hdl)hdl);
+    result = (int)gpi_get_object_type((gpi_sim_hdl)hdl);
```

Comment:
I think this would be clearer if you declared `result` to be the correct type, and added the cast on the line below.